src/main/java/core/basesyntax/Car.java

```diff
@@ -1,5 +1,45 @@
 package core.basesyntax;
 
-public class Car {
+public class Car implements Cloneable {
 
+    private final String name;
+    private final String model;
+    private final int weight;
+    private final int maxSpeed;
+    private final int price;
+    private Engine engine;
+
+    public String getName() {
+        return name;
+    }
+
+    public String getModel() {
+        return model;
+    }
+
+    public int getWeight() {
+        return weight;
+    }
+
+    public int getMaxSpeed() {
+        return maxSpeed;
+    }
+
+    public int getPrice() {
+        return price;
+    }
+
+    public Car(String name, String model, int weight, int maxSpeed, int price, Engine engine) {
+        this.name = name;
+        this.model = model;
+        this.weight = weight;
+        this.maxSpeed = maxSpeed;
+        this.price = price;
+        this.engine = engine;
+    }
+
+    @Override
+    protected Object clone() throws CloneNotSupportedException {
+        return super.clone();
```

Comment:
@Valdes2 
what if you will do next?

    
public static void main(String[] args) throws CloneNotSupportedException {
        Plane plane = new Plane();
        plane.setName("hello!");
        plane.setEngine(new Engine("type 1"));
        Plane clone = (Plane) plane.clone();
        clone.setName("buy-buy");
        clone.getEngine().setType("type 2");
        System.out.println(plane);
        System.out.println(clone);
    }

Reply:
fixed